src/app/loadout-drawer/subclass-drawer/SubclassPlugDrawer.tsx

```diff
@@ -32,40 +32,47 @@ export default function SubclassPlugDrawer({
   const language = useSelector(languageSelector);
   const defs = useD2Definitions();
   const profileResponse = useSelector(profileResponseSelector);
-  const maxAspects = subclass.sockets
-    ? getSocketsByCategoryHash(subclass.sockets, SocketCategoryHashes.Aspects).length
-    : 0;
-
-  const { initiallySelected, plugs, aspects, fragments, sortPlugs, sortPlugGroups } =
-    useMemo(() => {
-      const initiallySelected = Object.values(socketOverrides)
-        .map((hash) => defs?.InventoryItem.get(hash))
-        .filter(isPluggableItem);
-      const { plugs, aspects, fragments } = getPlugsForSubclass(defs, profileResponse, subclass);
-
-      // Return these as an array as that is what the plug drawer uses
-      const plugsToReturn = Array.from(plugs);
-
-      const sortPlugs = compareBy((plug: PluggableInventoryItemDefinition) =>
-        plugsToReturn.indexOf(plug)
-      );
-
-      // The grouping we use in the plug drawer breaks the plug ordering, this puts the groups in the
-      // correct order again as we build the set of plugs by iterating the categories in order
-      const sortPlugGroups = compareBy(
-        (group: PluggableInventoryItemDefinition[]) =>
-          group.length && plugsToReturn.indexOf(group[0])
-      );
-      return {
-        initiallySelected,
-        plugs: plugsToReturn,
-        aspects,
-        fragments,
-        sortPlugs,
-        sortPlugGroups,
-      };
-    }, [defs, profileResponse, socketOverrides, subclass]);
 
+  const {
+    initiallySelected,
+    plugsWithMaxSelectableSets,
+    aspects,
+    fragments,
+    sortPlugs,
+    sortPlugGroups,
+  } = useMemo(() => {
+    const initiallySelected = Object.values(socketOverrides)
+      .map((hash) => defs?.InventoryItem.get(hash))
```

Comment:
How would defs be undefined?

Reply:
It's just the type the hook returns and vscode adding it automatically. We need to run the `isPluggableItem` over it anyway so I will assert its not null but it makes no difference.